google/cloud/storage/credentials_test.cc

```diff
@@ -0,0 +1,95 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/credentials.h"
+#include "google/cloud/internal/setenv.h"
+#include "google/cloud/storage/internal/authorized_user_credentials.h"
+#include "google/cloud/storage/internal/service_account_credentials.h"
+#include "google/cloud/testing_util/environment_variable_restore.h"
+#include <gmock/gmock.h>
+#include <fstream>
+#include <typeinfo>
+
+namespace google {
+namespace cloud {
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
+namespace {
+
+char const VAR_NAME[] = "GOOGLE_APPLICATION_CREDENTIALS";
+
+class CredentialsTest : public ::testing::Test {
+ protected:
+  void SetUp() override { env_.SetUp(); }
+  void TearDown() override { env_.TearDown(); }
+
+ private:
+  testing_util::EnvironmentVariableRestore env_ =
+      testing_util::EnvironmentVariableRestore(VAR_NAME);
+};
+
+TEST_F(CredentialsTest, Insecure) {
+  InsecureCredentials credentials;
+  EXPECT_EQ("", credentials.AuthorizationHeader());
+}
+
+TEST_F(CredentialsTest, LoadAuthorizedUser) {
+  char const filename[] = "authorized-user.json";
+  std::ofstream os(filename);
+  os << R"""({
+  "client_id": "test-invalid-test-invalid.apps.googleusercontent.com",
```

Comment:
Did you mean this client_id string to match either the test name or code file name? Was this copy/pasted from somewhere else?

Reply:
No.  I just wanted it to look like client_id strings usually do, but clearly marked as invalid.